scripts/helpers.sh

```diff
@@ -33,9 +33,26 @@ battery_status() {
 	elif command_exists "upower"; then
 		# sort order: attached, charging, discharging
 		for battery in $(upower -e | grep battery); do
-			upower -i $battery | grep state | awk '{print $2}'
+			upower -i "$battery" | grep state | awk '{print $2}'
 		done | sort | head -1
 	elif command_exists "acpi"; then
 		acpi -b | grep -oi 'discharging' | awk '{print tolower($0)}'
 	fi
 }
+
+ac_online() {
+	local toggle_plug_status="$(get_tmux_option "@batt_toggle_plug_status" "false")"
+	[[ "${toggle_plug_status}" == "false" ]] && return 1
+
+	if command_exists "upower"; then
+		[[ "$(upower -i /org/freedesktop/UPower/devices/line_power_AC | awk -F: '/online/ {gsub(/ /, "", $NF); print $NF}')" == "yes" ]]
+	elif command_exists "acpi"; then
+		acpi -a | grep -q 'on-line'
+	elif command_exists "pmset"; then
+		[[ $(pmset -g | awk '/^AC Power/ {gsub(/[^*]/, "", $NF); print $NF}') == '*' ]]
```

Comment:
Not sure which version of Mac OS X that was on the laptop I tried on, but the output was as follows:

    MacBook-Pro:~ mgor$ pmset -g | awk '/Power/'
    Battery Power           -1
    AC Power                -1*

Which is why I didn't have `pmset -g batt`.

But with your output, I would suggest to chain that `elif`-case to:

    pmset -g batt | egrep -q "^Now draining from.*AC Power"

(I hope that `egrep` is available in Mac OS X)